nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,467 @@
+    info['colorscale'] = colors
+    return info
+
+
+def _fill_html_template(info, embed_js=True):
+    as_json = json.dumps(info)
+    as_html = get_html_template('surface_plot_template.html').replace(
+        'INSERT_STAT_MAP_JSON_HERE', as_json)
+    as_html = add_js_lib(as_html, embed_js=embed_js)
+    return SurfaceView(as_html)
+
+
+def view_img_on_surf(stat_map_img, mesh='fsaverage5',
+                     threshold=None, cmap=cm.cold_hot,
+                     black_bg=False):
+    """
+    Insert a surface plot of a statistical map into an HTML page.
+
+    Parameters
+    ----------
+    stat_map_img : Niimg-like object, 3d
+        See http://nilearn.github.io/manipulating_images/input_output.html
+
+    mesh : str or dict, optional.
+        if 'fsaverage5', use fsaverage5 mesh from nilearn.datasets
+        if 'fsaverage', use fsaverage mesh from nilearn.datasets
+        if a dictionary, keys should be 'infl_left', 'pial_left', 'sulc_left',
+        'infl_right', 'pial_right', 'sulc_right', containing inflated and
+        pial meshes, and sulcal depth values for left and right hemispheres.
+
+    threshold : str, number or None, optional (default=None)
+        If None, no thresholding.
+        If it is a number only values of amplitude greater
+        than threshold will be shown.
+        If it is a string it must finish with a percent sign,
+        e.g. "25.3%", and only values of amplitude above the
+        given percentile will be shown.
+
+    cmap : str or matplotlib colormap, optional
+
+    black_bg : bool, optional (default=False)
+
+    Returns
+    -------
+    SurfaceView : plot of the stat map.
+        It can be saved as an html page or rendered (transparently) by the
+        Jupyter notebook. Useful methods are :
+
+        - 'resize' to resize the plot displayed in a Jupyter notebook
+        - 'save_as_html' to save the plot to a file
+        - 'open_in_browser' to save the plot and open it in a web browser.
+
+    See Also
+    --------
+    nilearn.plotting.view_surf: plot from a surface map on a cortical mesh.
+
+    """
+    info = full_brain_info(
+        volume_img=stat_map_img, mesh=mesh, threshold=threshold,
+        cmap=cmap, black_bg=black_bg)
+    return _fill_html_template(info, embed_js=True)
+
+
+def view_surf(surf_mesh, surf_map=None, bg_map=None, threshold=None,
+              cmap=cm.cold_hot, black_bg=False, symmetric_cmap=True):
+    """
+    Insert a surface plot of a surface map into an HTML page.
+
+    Parameters
+    ----------
+    surf_mesh: str or list of two numpy.ndarray
+        Surface mesh geometry, can be a file (valid formats are
+        .gii or Freesurfer specific files such as .orig, .pial,
+        .sphere, .white, .inflated) or
+        a list of two Numpy arrays, the first containing the x-y-z coordinates
+        of the mesh vertices, the second containing the indices
+        (into coords) of the mesh faces.
+
+    surf_map: str or numpy.ndarray, optional.
```

Comment:
This would not work when I used one ROI to plot on surface. For example, pcc labels from plot_surf_stat_map example.

Reply:
```python
ValueError: Mismatch between number of nodes in mesh (10242) and size of surface data (66)
```

Reply:
Thanks. Worth adding this into the user guide if you think we are not cluttering too much there ?

Reply:
I think maybe the snippet is a bit long, and plotting one roi or plotting an atlas is pretty much the same isn't it? I would just leave the example with the atlas in the user guide, and fix the interface of plot_surf_roi. what do you think?

Reply:
see #1697